### PR TITLE
Bluetooth: Audio: Make VCS optional notify characteristics optional

### LINKED ISSUE #57476: LE Audio: VCS optional support Volume Flags notifications
**Is your enhancement proposal related to a problem? Please describe.**
The Volume Flags are only optionally notifiable and the implementation should reflect that. 

**Describe the solution you'd like**
The server needs to optionally support the notifications of the volume flags. 
The Client needs to optionally support subscribing to and receiving the volume flag notifications. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by fredrikdanebjer**
This change makes VCS Volume Flag characteristic Notify property optional and selectable through Kconfig.

closes #57476

### CODE REVIEW

File: subsys/bluetooth/audio/vcp_vol_rend.c [hunk 1]
```diff
@@ -192,9 +192,11 @@ static ssize_t write_vcs_control(struct bt_conn *conn,
 	if (volume_change && !vol_rend.flags) {
 		vol_rend.flags = 1;
 
+#if defined(CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE)
 		bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
 				    vol_rend.service_p->attrs,
 				    &vol_rend.flags, sizeof(vol_rend.flags));
+#endif /* CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE */
```
Comment (Thalley) [inline subsys/bluetooth/audio/vcp_vol_rend.c:199]: Since this is not using any guarded values, suggest to use `IS_ENABLED` instead

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/vcp_vol_rend.c (hunk 1 above)
Comment (fredrikdanebjer) [reply subsys/bluetooth/audio/vcp_vol_rend.c:199]: What do you mean by guarded values?
I just read the FAQ on IS_ENABLED and I understand the benefits - but when should we not use it?
Or is it beneficial to just replace all `defined` with `IS_ENABLED`?

Comment (Thalley) [reply subsys/bluetooth/audio/vcp_vol_rend.c:199]: Doing 
```c
	if (IS_ENABELD(CONFIG_FOO)) {
		/* something */
	}
```
is preferred over 
```c
#if defined(CONFIG_FOO)
	/* something */
#endif
```

But if `/* something */` contains access to static functions that are guarded by `CONFIG_FOO` (i.e. not compiled) or if it contains access to struct members that are also guarded by `CONFIG_FOO` then we cannot use `IS_ENABLED`. 

In this case here, everything in 
```c
		bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
				    vol_rend.service_p->attrs,
				    &vol_rend.flags, sizeof(vol_rend.flags));
```
is always compiled, so we can use IS_ENABLED

File: subsys/bluetooth/audio/vcp_vol_rend.c [hunk 2]
```diff
@@ -192,9 +192,11 @@ static ssize_t write_vcs_control(struct bt_conn *conn,
 	if (volume_change && !vol_rend.flags) {
 		vol_rend.flags = 1;
 
+#if IS_ENABLED(CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE)
 		bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
 				    vol_rend.service_p->attrs,
 				    &vol_rend.flags, sizeof(vol_rend.flags));
+#endif /* CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE */
```
Comment (Thalley) [inline subsys/bluetooth/audio/vcp_vol_rend.c:199]:
```suggestion
		if(IS_ENABLED(CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE)) {
			bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
					    vol_rend.service_p->attrs,
					    &vol_rend.flags, sizeof(vol_rend.flags));
		}
```

`IS_ENABLED` is only an advantage if you put it in an `if` and not an `#if`

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/vcp_vol_rend.c [hunk 3]
```diff
@@ -192,9 +192,11 @@ static ssize_t write_vcs_control(struct bt_conn *conn,
 	if (volume_change && !vol_rend.flags) {
 		vol_rend.flags = 1;
 
-		bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
-				    vol_rend.service_p->attrs,
-				    &vol_rend.flags, sizeof(vol_rend.flags));
+		if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND_VOL_FLAGS_NOTIFIABLE)) {
+			bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
+						vol_rend.service_p->attrs,
+						&vol_rend.flags, sizeof(vol_rend.flags));
```
Comment (Thalley) [inline subsys/bluetooth/audio/vcp_vol_rend.c:198]: Indentation seems off here :) 

Should be

```suggestion
			bt_gatt_notify_uuid(NULL, BT_UUID_VCS_FLAGS,
					    vol_rend.service_p->attrs,
					    &vol_rend.flags, sizeof(vol_rend.flags));
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor code style fix, otherwise LGTM

**Review by MariuszSkamra — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**